### PR TITLE
Docker-compose metric collection

### PR DESCRIPTION
**Pull request by optimisticben**
**Description**
Adding metric collection to the local docker-compose project.

The included README has additional information.

### CODE REVIEW

File: ops/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+Grafana requires a login. The defaults are:
+```
+user: admin
+password: optimism
+```
+
+## Data persistance
+
+Grafana data is not currently saved. Any modifications or additions will be lost on container restart.
```
Comment (gakonst) [inline ops/README.md:49]: Should it be?

File: ops/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+InfluxDB is persisting data to a Docker volume.
+
+**Stopping the project removing the containers will not clear this volume**
+
+To remove the influxdb data, run a command like
+```
+docker volume rm ops_influxdeb_data
```
Comment (gakonst) [inline ops/README.md:57]: Typo?

```suggestion
docker volume rm ops_influxdb_data
```

File: ops/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+```
+Or, after stopiing the project, clean up all unused containers and volumes.
```
Comment (gakonst) [inline ops/README.md:59]:
```suggestion
Or, after stopping the project, clean up all unused containers and volumes.  
```

File: ops/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+**Stopping the project removing the containers will not clear this volume**
+
+To remove the influxdb data, run a command like
+```
+docker volume rm ops_influxdeb_data
+```
+Or, after stopiing the project, clean up all unused containers and volumes.  
+
+**This step remove way more then just this project, please make sure you know what you're doing**
```
Comment (gakonst) [inline ops/README.md:61]:
```suggestion
**This step removes way more then just this project, please make sure you know what you're doing**
```

File: ops/README.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+```
+docker system prune --volumes
```
Comment (gakonst) [inline ops/README.md:63]: Maybe we should just remove this suggestion? I can see somebody running it by accident..

File: ops/docker-compose-metrics.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  influxdb:
+    image: quay.io/influxdb/influxdb:1.6
+    env_file:
+      -  ./envs/metrics.env
+    volumes:
+      - influxdeb_data:/root/.influxdb
+
+volumes:
+  influxdeb_data:
```
Comment (gakonst) [inline ops/docker-compose-metrics.yml:24]:
```suggestion
  influxdb_data:
```

File: ops/docker-compose.yml [hunk 7]
```diff
@@ -11,26 +11,26 @@ services:
       dockerfile: ./ops/docker/Dockerfile.monorepo
 
   # this is a helper service used because there's no official hardhat image
-  l1_chain:
+  l1geth:
```
Comment (gakonst) [inline ops/docker-compose.yml:14]: Can we keep it as `l1_chain`? Hardhat is not a geth instance.

File: ops/docker/grafana/provisioning/dashboards/geth.json [hunk 8]
```diff
@@ -0,0 +1,19870 @@
+{
```
Comment (gakonst) [inline ops/docker/grafana/provisioning/dashboards/geth.json]: Does this file need to be committed?

**Review by gakonst — CHANGES_REQUESTED** [review]
LGTM generally, minor comments before approving.

File: ops/README.md (hunk 1 above)
Comment (snario) [reply ops/README.md:49]: I think it's fine for local development to not be saved; you can be mindful as a developer about keeping the container around if you need it.

File: ops/docker-compose.yml (hunk 7 above)
Comment (snario) [reply ops/docker-compose.yml:14]: Maybe we should just call it `hardhat_l1_network` to be explicit

File: ops/README.md (hunk 1 above)
Comment (optimisticben) [reply ops/README.md:49]: Easy enough to add and it should prevent some unexpected behavior (who reads docs?).

I'll add this volume.

File: ops/README.md (hunk 5 above)
Comment (optimisticben) [reply ops/README.md:63]: Yeah, I wasn't 💯 on this, heh. Removing.

File: ops/docker-compose.yml (hunk 7 above)
Comment (optimisticben) [reply ops/docker-compose.yml:14]: Yeah, I'll revert. nbd.

File: ops/docker/grafana/provisioning/dashboards/geth.json (hunk 8 above)
Comment (optimisticben) [reply ops/docker/grafana/provisioning/dashboards/geth.json]: Not a lot of great choices as this dashboard is coming from an external source.

What I could come up with:

Have the user log into grafana and import the dashboard from https://grafana.com/grafana/dashboards/13877
Have another container fetch, template and save the file to the provisioning path
Have another container fetch, template and POST it to the grafana api

File: ops/README.md (hunk 1 above)
Comment (optimisticben) [reply ops/README.md:49]: Grafana volume added.

**Comment by tynes** [comment]
A newer dashboard for geth was created here: https://gist.github.com/karalabe/1e26f9ea5c842fb118584edadc454e18
Its supposed to work with version 1.10.3, I do not think it will completely work with our version. Posting this here as more of a way to keep track of the new dashboard

**Comment by tynes** [comment]
I'm not particularly opposed to renaming `l1_chain` due to it being inconsistent with `l2geth` - no underscore. The `l1_chain` may not be geth under the hood while `l2geth` is always `l2geth` at this point. I do believe that `l1_chain` is built specifically as part of CI so name changes would need updates to places where `docker-compose build ...` is called. It seems like the name update was reverted and I don't want to bikeshed but just saying that I'd support a rename to `l1chain` for consistency purposes

**Comment by optimisticben** [comment]
> A newer dashboard for geth was created here: https://gist.github.com/karalabe/1e26f9ea5c842fb118584edadc454e18
> Its supposed to work with version 1.10.3, I do not think it will completely work with our version. Posting this here as more of a way to keep track of the new dashboard

Nice. I just grabbed this one from the ethereum docs. I'll get it updated, or added as well.

**Comment by optimisticben** [comment]
I reverted the `l1_chain` changes since they really had nothing to do with the PR.

There are some other nitpicks I can bundle in another PR.

File: ops/docker/grafana/provisioning/dashboards/geth.json (hunk 8 above)
Comment (optimisticben) [reply ops/docker/grafana/provisioning/dashboards/geth.json]: Removed the dashboard from the repo in favor of a script to pull in a list of dashboards and share them with grafana though a Docker volume.

This list is just in the script for simplicity, and includes the dashboard link @tynes referenced.

**Comment by optimisticben** [comment]
Updated to include Prometheus collection since the DTL endpoint is available. A dashboard for the service will be in a follow up PR.

**Pull request merged**